flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_VerticalRevisionPage.js

```diff
@@ -48,73 +74,150 @@ class CDUVerticalRevisionPage {
                 [speedConstraint ? speedConstraint + "[color]magenta" : "*[\xa0\xa0\xa0][color]cyan", altitudeConstraint ? altitudeConstraint + "[color]magenta" : "[\xa0\xa0\xa0\xa0]*[color]cyan"],
                 ["MACH/START WPT[color]inop", ""],
                 [`\xa0{inop}[\xa0]/{small}${waypointIdent}{end}{end}`, ""],
                 [""],
                 ["<WIND", "STEP ALTS>[color]inop"],
                 [""],
-                ["<RETURN"]
+                [confirmConstraint ? "{amber}*CLB{end}" : "<RETURN", confirmConstraint ? "{amber}DES*{end}" : "", confirmConstraint ? "{amber}{small}OR{end}{end}" : ""]
             ]);
             mcdu.onLeftInput[0] = () => {}; // EFOB
             mcdu.onRightInput[0] = () => {}; // EXTRA
-            mcdu.onLeftInput[1] = () => {}; // CLB SPD LIM
-            mcdu.onRightInput[1] = () => {}; // RTA
-            mcdu.onLeftInput[2] = async (value, scratchpadCallback) => {
-                const speed = (value !== FMCMainDisplay.clrValue) ? parseInt(value) : 0;
-                if (isFinite(speed)) {
-                    if (speed >= 0) {
-                        mcdu.flightPlanManager.setWaypointSpeed(speed, mcdu.flightPlanManager.indexOfWaypoint(waypoint), () => {
-                            mcdu.updateConstraints();
-                            this.ShowPage(mcdu, waypoint);
-                        });
+            mcdu.onLeftInput[1] = (value, scratchpadCallback) => {
+                if (mcdu.flightPhaseManager.phase === FmgcFlightPhases.CRUISE || mcdu.flightPhaseManager.phase > FmgcFlightPhases.APPROACH) {
+                    return;
+                }
+
+                const climb = mcdu.flightPhaseManager.phase < FmgcFlightPhases.CRUISE;
+                if (value === FMCMainDisplay.clrValue) {
+                    if (climb) {
+                        if (mcdu.climbSpeedLimitPilot) {
+                            mcdu.climbSpeedLimit = 250;
+                            mcdu.climbSpeedLimitAlt = 10000;
+                        } else {
+                            mcdu.climbSpeedLimit = undefined;
+                            mcdu.climbSpeedLimitAlt = undefined;
+                        }
+                        mcdu.climbSpeedLimitPilot = false;
+                    } else {
+                        if (mcdu.climbSpeedLimitPilot) {
+                            mcdu.descentSpeedLimit = 250;
+                            mcdu.descentSpeedLimitAlt = 10000;
+                        } else {
+                            mcdu.descentSpeedLimit = undefined;
+                            mcdu.descentSpeedLimitAlt = undefined;
+                        }
+                        mcdu.descentSpeedLimitPilot = false;
                     }
+                    CDUVerticalRevisionPage.ShowPage(mcdu, waypoint);
+                    return;
+                }
+
+                const m = value.match(/^([0-9]{1,3})\/(((FL)?([0-9]{1,3}))|([0-9]{4,5}))$/);
```

Comment:
It's a tightly scoped local var required because a) Coherent doesn't support named sub-patterns, and b) JS can't destructure null to an array... so we first have to assign the match result to such a var, check for null, and then destructure it. Generally the length of my var names are proprotionate to the size of their scope/lifetime :D. I could rename it to match or matchResult I guess.